python/lsst/ip/isr/isrFunctions.py

```diff
@@ -592,6 +594,230 @@ def brighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain, ga
+        kernel K.
+    fStep: `numpy.array`
+        The array of flux values which act as the source of the flux transfer.
+    correctionMode: `bool`
+        Defines if applying correction (True) or generating sims (False).
+
+    Returns
+    -------
+    corr:
+        BFE correction array
+    """
+
+    if cFunc.shape != fStep.shape:
+        raise RuntimeError(f'transferFlux: array shapes do not match: {cFunc.shape}, {fStep.shape}')
+
+    # set the sign of the correction and set its value for the
+    # time averaged solution
+    if correctionMode:
+        # negative sign if applying BFE correction
+        factor = -0.5
+    else:
+        # positive sign if generating BFE simulations
+        factor = 0.5
+
+    # initialise the BFE correction image to zero
+    corr = numpy.zeros_like(cFunc)
+
+    # Generate a 2D mesh of x,y coordinates
+    yDim, xDim = cFunc.shape
+    y = numpy.arange(yDim, dtype=int)
+    x = numpy.arange(xDim, dtype=int)
+    xc, yc = numpy.meshgrid(x, y)
+
+    # process each axis in turn
+    for ax in [0, 1]:
+
+        # gradient of phi on right/upper edge of pixel
+        diff = numpy.diff(cFunc, axis=ax)
+
+        # expand array back to full size with zero gradient at the end
+        gx = numpy.zeros_like(cFunc)
+        yDiff, xDiff = diff.shape
+        gx[:yDiff, :xDiff] += diff
+
+        # select pixels with either positive gradients on the right edge,
+        # flux flowing to the right/up
+        # or negative gradients, flux flowing to the left/down
+        for i, sel in enumerate([gx > 0, gx < 0]):
+            xSelPixels = xc[sel]
+            ySelPixels = yc[sel]
+            # and add the flux into the pixel to the right or top
+            # depending on which axis we are handling
+            if ax == 0:
+                xPix = xSelPixels
+                yPix = ySelPixels+1
+            else:
+                xPix = xSelPixels+1
+                yPix = ySelPixels
+            # define flux as the either current pixel value or pixel
+            # above/right
+            # depending on whether positive or negative gradient
+            if i == 0:
+                # positive gradients, flux flowing to higher coordinate values
+                flux = factor * fStep[sel]*gx[sel]
+            else:
+                # negative gradients, flux flowing to lower coordinate values
+                flux = factor * fStep[yPix, xPix]*gx[sel]
+            # change the fluxes of the donor and receiving pixels
+            # such that flux is conserved
+            corr[sel] -= flux
+            corr[yPix, xPix] += flux
+
+    # return correction array
+    return corr
+
+
+def fluxConservingBrighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain,
+                                           gains=None, correctionMode=True):
+    """Apply brighter fatter correction in place for the image.
+    Modified version conserves flux, with improved correction of PSF core.
```

Comment:
I think this docstring needs to be reworded a bit.  I'd move this line and the next to a new paragraph, switching "modified" for "this", such that
```
This version presents a modified version of the algorithm found in ``lsst.ip.isr.isrFunctions.brighterFatterCorrection`` which conserves the image flux, resulting in improved correction of the cores of stars.  The convolution has also been modified to mitigate edge effects.
```